### PR TITLE
Code Quality: Introduce details layout columns generator phase 1

### PR DESCRIPTION
**Pull request by 0x5bfa**
## Summary

As of now, you have to add literally 250 lines only for one column when you are adding a new one (you can see [git column introduction PR](https://github.com/files-community/Files/pull/12485)). We are introducing columns generator to mitigate that situation.

## Task Checklist

- [x] Clean up
- [ ] Introduce details layout columns generator

## Steps To Validate Changes

1. Launch the app
2. Go to a folder
3. Resize some columns
4. Hide some columns
5. Go to another folder
6. See the columns you hid are shown in that folder
7. Go to git folder
8. See git columns appear

If nothing happen, you confirmed layout settings for each page are stored correctly in user_settings.db, hide/show operation is working properly, the app is able to fetch settings from the database and use them in an actual layout.

## PR Checklist

- [x] **Development**: Related: #12505
- [x] **Approval**: Have discussed with and got approval from the team[^1]
- [x] **Tests**: Tested accessibility on the local end
- [x] **Deployment**: Deployed the app on the local end
- [ ] **Localization**: Modified en-US string resources[^2]
- [ ] **Co-authors**: _N/A_

## Screenshots

Related codebase structure:
image

[^1]: If the request hasn't been agreed by the team, this work might be rejected. Make sure that you get approval from the team before opening any PR related the request.
[^2]: If you removed any en-US string resources, make sure that there are no references of those resources. When you add a new en-US string resources, do not make any changes on other languages' resources; [Crowdin](https://crowdin.com/project/files-app) will do that, instead.

### CODE REVIEW

File: src/Files.App/Views/Layouts/DetailsLayoutPage.xaml.cs
Comment (yaira2) [inline src/Files.App/Views/Layouts/DetailsLayoutPage.xaml.cs:582]: I think this is needed for when the window is resized

Comment (0x5bfa) [reply src/Files.App/Views/Layouts/DetailsLayoutPage.xaml.cs:582]: This method does nothing

```cs
		public void SetDesiredSize(double width)
		{
			if (TotalWidth > width || TotalWidth < width)
			{
				var proportion = width / TotalWidth;

				//SetColumnSizeProportionally(proportion);
			}
		}
```

Comment (yaira2) [reply src/Files.App/Views/Layouts/DetailsLayoutPage.xaml.cs:582]: Can you try to find out when this was commented out? It'll be good to know why we originally had this in place.

Comment (0x5bfa) [reply src/Files.App/Views/Layouts/DetailsLayoutPage.xaml.cs:582]: LOL. That was you.
https://github.com/files-community/Files/commit/976484fbe267f64025287f7602916f477867fd05

Comment (yaira2) [reply src/Files.App/Views/Layouts/DetailsLayoutPage.xaml.cs:582]: Good fine! Looks like I left it in so we can add an option to auto size columns to fit.

Comment (0x5bfa) [reply src/Files.App/Views/Layouts/DetailsLayoutPage.xaml.cs:582]: I'm not sure if we should leave as is because I'm going to introduce

Comment (0x5bfa) [reply src/Files.App/Views/Layouts/DetailsLayoutPage.xaml.cs:582]: If we will remove it and find out we need it, I will back it back soon in another PR that will be open as soon as this PR merged.

**Comment by 0x5bfa** [comment]
Loooks like renaming linking on diff view become broken because a lot of reordering changes were made and git no longer recognize LayoutPreferencesManager was FolderSettingsViewModel. Please keep in mind this PR only contains reordering & removal of null warnings refactoring.

**Review by yaira2 — APPROVED** [review]

**Comment by yaira2** [comment]
Looks good behavior wise. I'd like to wait for an additonal review from @gave92 or @hishitetsu before merging.

File: src/Files.App/Data/Items/DetailsLayoutColumnItem.cs
Comment (hishitetsu) [inline src/Files.App/Data/Items/DetailsLayoutColumnItem.cs:1]: `GetHashCode()` can be improved in the same way as `LayoutPreferencesItem`.

**Review by hishitetsu — APPROVED** [review]
Great work! LGTM

File: src/Files.App/Data/Items/DetailsLayoutColumnItem.cs
Comment (0x5bfa) [reply src/Files.App/Data/Items/DetailsLayoutColumnItem.cs:1]: I'm gonna fix in the stage 2 PR. Thanks!

**Pull request merged**
